Specifications/QIR/Reserved-Operations.md

```diff
@@ -0,0 +1,41 @@
+| T | [`__quantum__qis__t__body`](https://github.com/microsoft/qsharp-runtime/blob/5a379fd1c72ea19cb50d6acc501c6896894e96ac/src/Qir/Runtime/lib/QSharpCore/qsharp-core-qis.ll#L224) | The [T operation](https://docs.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.intrinsic.t) | [Single-qubit operations](https://docs.microsoft.com/en-us/azure/quantum/concepts-the-qubit#single-qubit-operations), [Rotations](https://docs.microsoft.com/en-us/azure/quantum/user-guide/libraries/standard/prelude#rotations). ![T = \begin{bmatrix} 1 & 0 \\ 0 & e^{i\pi/4} \\ \end{bmatrix} = \begin{bmatrix} 1 & 0 \\ 0 & \frac{1}{\sqrt{2}}+ \frac{1}{\sqrt{2}}i \\ \end{bmatrix}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+T+%3D+%5Cbegin%7Bbmatrix%7D+1+%26+0+%5C%5C+0+%26+e%5E%7Bi%5Cpi%2F4%7D+%5C%5C+%5Cend%7Bbmatrix%7D+%3D+%0A%5Cbegin%7Bbmatrix%7D+1+%26+0+%5C%5C+0+%26+%5Cfrac%7B1%7D%7B%5Csqrt%7B2%7D%7D%2B+%5Cfrac%7B1%7D%7B%5Csqrt%7B2%7D%7Di+%5C%5C+%5Cend%7Bbmatrix%7D) |
+
+| Operation Name | QIR Implementation | Description | Notes |
+|----------------|-----------------|-------------|-------|
+| Tdg | [`__quantum__qis__t__adj`](https://github.com/microsoft/qsharp-runtime/blob/5a379fd1c72ea19cb50d6acc501c6896894e96ac/src/Qir/Runtime/lib/QSharpCore/qsharp-core-qis.ll#L230) | The adjoint [T operation](https://docs.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.intrinsic.t) or T Dagger (![T^\dagger](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+T%5E%5Cdagger)) | ![T^\dagger = \begin{bmatrix} 1 & 0 \\ 0 & e^{-i\pi/4} \\ \end{bmatrix}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+T%5E%5Cdagger+%3D+%5Cbegin%7Bbmatrix%7D+1+%26+0+%5C%5C+0+%26+e%5E%7B-i%5Cpi%2F4%7D+%5C%5C+%5Cend%7Bbmatrix%7D) (see ![T^\dagger](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+T%5E%5Cdagger) [here](https://en.wikipedia.org/wiki/Clifford_gates#Building_a_universal_set_of_quantum_gates)) |
+| S | [`__quantum__qis__s__body`](https://github.com/microsoft/qsharp-runtime/blob/5a379fd1c72ea19cb50d6acc501c6896894e96ac/src/Qir/Runtime/lib/QSharpCore/qsharp-core-qis.ll#L198) | The [S operation](https://docs.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.intrinsic.s) (Phase Gate)  | [Single-qubit operations](https://docs.microsoft.com/en-us/azure/quantum/concepts-the-qubit#single-qubit-operations), [Other Single-Qubit Cliffords](https://docs.microsoft.com/en-us/azure/quantum/user-guide/libraries/standard/prelude#other-single-qubit-cliffords). ![S = T^2](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+S+%3D+T%5E2) |
```

Comment:
The algebraic relationships between the operations aren't really necessary. The operation matrices specify the operations completely.

Reply:
I assume that with "algebraic relationships" you mean the fragments like this "S = T^2".
I explained that in the [earlier comment](https://github.com/microsoft/qsharp-language/pull/105#discussion_r651227848).

Please feel free to let me know if you would prefer me to remove those fragments.

Reply:
Personally I would leave them out. It is unlikely to be useful to a hardware vendor (since the way they implement these operations is unlikely to depend on the algebraic relations) and doesn't add precision. The matrices provide exactness, everything else is extraneous.